components/style/stylearc.rs

```diff
@@ -71,6 +71,40 @@ pub struct Arc<T: ?Sized> {
     ptr: *mut ArcInner<T>,
 }
 
+/// An Arc that is known to be uniquely owned
+///
+/// This lets us build arcs that we can mutate before
+/// freezing, without needing to change the allocation
+pub struct UniqueArc<T: ?Sized>(Arc<T>);
```

Comment:
Meh, actually, I guess it's fine, and being here allows using the private member without unsafety or exposing extra stuff, so...